src/ports/gelato/components.ts

```diff
@@ -0,0 +1,169 @@
+  ): Promise<string> {
+    const gelatoAPIKey = await config.requireString('GELATO_API_KEY')
+    const gelatoAPIURL = await config.requireString('GELATO_API_URL')
+    const chainId = await config.requireNumber('COLLECTIONS_CHAIN_ID')
+
+    const response = await fetcher.fetch(
+      `${gelatoAPIURL}/relays/v2/sponsored-call`,
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          chainId,
+          target: transactionData.params[0],
+          data: transactionData.params[1],
+          sponsorApiKey: gelatoAPIKey,
+        }),
+      }
+    )
+
+    if (response.ok) {
+      const data = (await response.json()) as { taskId: string }
+      return getTxHashFromGelatoResponse(data.taskId)
+    } else {
+      const message = await extractErrorMessage(response)
+
+      metrics.increment('dcl_error_service_errors_gelato')
+      logger.error(
+        `Gelato failed to relay the transaction with a ${response.status} status: ${message}`
+      )
+      throw new RelayerError(response.status, message)
+    }
+  }
+
+  const getTxHashFromGelatoResponse = async (
+    taskId: string
+  ): Promise<string> => {
+    const gelatoAPIURL = await config.requireString('GELATO_API_URL')
+    const gelatoMaxStatusChecks = await config.requireNumber(
+      'GELATO_MAX_STATUS_CHECKS'
+    )
+    const gelatoSleepTimeBetweenChecks = await config.requireNumber(
+      'GELATO_SLEEP_TIME_BETWEEN_CHECKS'
+    )
+
+    let txHash: string | undefined
+    let checks: number = 0
+
+    while (!txHash) {
+      if (checks >= gelatoMaxStatusChecks) {
+        logger.error('Gelato task status checks limit reached')
+        metrics.increment('dcl_error_timeout_gelato')
+        throw new RelayerTimeout('The limit of status checks was reached')
+      }
+
+      const response = await fetcher.fetch(
+        `${gelatoAPIURL}/tasks/status/${taskId}`
+      )
+
+      if (response.ok) {
+        const data = (await response.json()) as GelatoTaskStatusResponse
+        if (data.task.taskState === TaskState.ExecReverted) {
+          logger.error(
+            `Gelato task ${taskId} reverted: ${data.task.lastCheckMessage}`
+          )
+          metrics.increment('dcl_error_reverted_transactions_gelato')
+          throw new InvalidTransactionError(
+            'Transaction reverted',
+            ErrorCode.EXPECTATION_FAILED
+          )
+        } else if (data.task.taskState === TaskState.Cancelled) {
+          logger.error(
+            `Gelato task ${taskId} cancelled: ${data.task.lastCheckMessage}`
+          )
+          metrics.increment('dcl_error_cancelled_transactions_gelato')
+          const noBalanceLeftInGasTankError =
+            data.task.lastCheckMessage?.includes('No available token balance')
+          const noTokensConfiguredInGasTankSetError =
+            data.task.lastCheckMessage?.includes('No available token balance')
```

Comment:
these 2 variables are checking for the same string, is that intended? if so, we could just use one, right?